project/index.mjs

```diff
@@ -138,6 +138,94 @@ export function respondWithProject(req, res, project) {
   res.location(id).status(200).send(retVal)
 }
 
+/**
+ * Check for valid project keys and create defaults, then send to database
+ * @param {Object} req The request object
+ * @param {Object} res The response object
+ */
+async function createNewProject(req, res) {
+  let project = req.body
+  /* 
+  TODO:
+    - Add creator based on authenticated user
+    - Add group
+    - Add slug (and figure out a good way to format it)
+    - Check for manifest, create one in the database if it doesn't exist
+  */
+
+  // TEMPORARY stubbing out required keys we need to generate
+  project.creator = "user"
+  project.group = "group"
+  project.slug = "slug"
+
+  // Required keys
+  if (project.created) {
+    if (parseInt(project.created) === NaN) {
+      utils.respondWithError(res, 400, 'Project key "created" must be a date in UNIX time')
+      return
+    }
+  } else {
+    utils.respondWithError(res, 400, 'Project must have key "created"')
+    return
+  }
+  if (project.license) {
+    if (typeof project.license !== 'string') {
+      utils.respondWithError(res, 400, 'Project key "license" must be a string')
+      return
+    }
+  } else {
+    project.license = "CC-BY"
+  }
+  if (project.title) {
+    if (typeof project.title !== 'string') {
+      utils.respondWithError(res, 400, 'Project key "title" must be a string')
+      return
+    }
+  } else {
+    project.title = "Project" + project.creator + project.created.toString()
+  }
+
+  // Optional keys
+  if (project.tools) {
+    if (!Array.isArray(project.tools)) {
+      utils.respondWithError(res, 400, 'Project key "tools" must be an array')
+      return
+    }
+  }
+  if (project.tags) {
+    if (!Array.isArray(project.tags)) {
+      utils.respondWithError(res, 400, 'Project key "tags" must be an array')
+      return
+    }
+    if (!project.tags.every(tag => typeof tag === 'string')) {
+      utils.respondWithError(res, 400, 'Project key "tags" must be an array of strings')
+      return
+    }
+  }
+
+  req.body["@type"] = "Project"
```

Comment:
I think this would be a part of the helper function that checks for the minimum requirements.  Sure, we can assume this is what they meant.  However if we do this and they post to this with a `"@type":"User"` it would probably be better to give them a RESTful message about how this is all wrong.